### PR TITLE
Utopia-770 Tab Security Personal Information

### PR DESCRIPTION
**Pull request by antsand**
# Description

This PR includes the following proposed change(s):

- {List all the changes, if possible add the jira ticket #}

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

File: src/frontend/src/components/public/PIAFormTabs/Security_Personal_Information/security-personal-info-interface.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+export interface ISecurityPersonalInformation {
+  digitalToolsAndSystems: {
+    toolsAndAssessment: {
+      involveDigitalToolsAndSystems: string | null;
```
Comment (binggaofw) [inline src/frontend/src/components/public/PIAFormTabs/Security_Personal_Information/security-personal-info-interface.ts:4]: I think the backend only accept string, may be we can remove **| nul**l part

File: src/frontend/src/components/public/PIAFormTabs/Security_Personal_Information/index.tsx [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+      additionalStrategies: '',
+    },
+  };
+
+  const [securityPersonalInformationForm, setSecurityPersonalInformationForm] =
+    useState<ISecurityPersonalInformation>(
+      pia.securityPersonalInformation || defaultState,
+    );
+
+  const stateChangeHandler = (value: any, nestedkey: string) => {
```
Comment (binggaofw) [inline src/frontend/src/components/public/PIAFormTabs/Security_Personal_Information/index.tsx:38]: name should be nestedKey

File: src/frontend/src/components/public/PIAFormTabs/Security_Personal_Information/index.tsx [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+          <div className="drop-shadow section__padding-inline bg-white section__padding-block section-border-radius">
+            <p>
+              <strong>
+                {
+                  Messages.FormElements.DigitalTools.InvolveDigitalTools
+                    .Question.en
+                }
+              </strong>
+            </p>
+            <div className="">
```
Comment (binggaofw) [inline src/frontend/src/components/public/PIAFormTabs/Security_Personal_Information/index.tsx:110]: this className has empty value, can be remove

File: src/frontend/src/components/public/PIAFormTabs/Security_Personal_Information/index.tsx [hunk 4]
```diff
@@ -0,0 +1,412 @@
...
+    }
+    piaStateChangeHandler(
+      securityPersonalInformationForm,
+      'securityPersonalInformation',
+    );
+  };
+
+  return (
+    <>
+      <div className="">
```
Comment (binggaofw) [inline src/frontend/src/components/public/PIAFormTabs/Security_Personal_Information/index.tsx:94]: same here ,empty className

**Review by kushal-arora-fw — COMMENTED** [review]
Looks good overall. Fixed some autosaving issues. The only other major issue I have is that the app sends the stale data for text area even when the conditional rendering does not show the element anymore. It should not send the stale value.

**Review by kushal-arora-fw — APPROVED** [review]

**Pull request merged**